src/ServiceBouncer/MainForm.cs

```diff
@@ -137,20 +141,62 @@ private void InstallClicked(object sender, EventArgs e)
         {
-            var systemServices = await Task.Run(() => ServiceController.GetServices().Where(service => service.DisplayName.IndexOf(toolStripFilterBox.Text, StringComparison.OrdinalIgnoreCase) >= 0));
-            services.Clear();
-            foreach (var model in systemServices.Select(service => new ServiceViewModel(service)).OrderBy(x => x.Name))
+            try
+            {
+                var systemServices = await Task.Run(() => ServiceController.GetServices(MachineHostname).Where(service => service.DisplayName.IndexOf(toolStripFilterBox.Text, StringComparison.OrdinalIgnoreCase) >= 0));
+                Connect();
+                services.Clear();
+
+                foreach (var model in systemServices.Select(service => new ServiceViewModel(service)).OrderBy(x => x.Name))
+                {
+                    services.Add(model);
+                }
+
+                SetTitle();
+            }
+            catch (Exception e)
             {
-                services.Add(model);
+                Disconnect();
+                MessageBox.Show($"Unable to retrieve the services from {toolStripConnectToTextBox.Text}. Message: {e.Message}", "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
             }
+        }
 
-            SetTitle();
+        private void Connect()
+        {
+            toolStripConnectButton.Text = "Disconnect";
+            toolStripConnectButton.ToolTipText = "Disconnect";
+            toolStripConnectButton.Tag = "Connected";
+            toolStripConnectButton.Enabled = true;
+            servicesDataGridView.Enabled = true;
+            toolStripConnectToTextBox.Enabled = true;
+            toolStripConnectButton.Image = Properties.Resources.Connected;
+            toolStripStatusLabel.Text = $"Connected to {MachineHostname}.";
+        }
+
+        private void Disconnect()
+        {
+            toolStripConnectButton.Text = "Connected";
+            toolStripConnectButton.ToolTipText = "Connected";
+            toolStripConnectButton.Tag = "Disconnect";
+            toolStripConnectButton.Image = Properties.Resources.Disconnected;
+            toolStripConnectButton.Enabled = true;
+            servicesDataGridView.Enabled = true;
+            toolStripConnectToTextBox.Enabled = true;
+            toolStripStatusLabel.Text = "Disconnected";
+            services.Clear();
         }
 
         private void SetTitle()
         {
-            var titles = services.GroupBy(s => s.Status).Select(s => (string.IsNullOrWhiteSpace(s.Key) ? "Unknown" : s.Key) + ": " + s.Count());
-            Text = "Total: " + services.Count + ", " + string.Join(", ", titles);
+            if (services.Any())
+            {
+                var titles = services.GroupBy(s => s.Status).Select(s => (string.IsNullOrWhiteSpace(s.Key) ? "Unknown" : s.Key) + ": " + s.Count());
+                Text = "Total: " + services.Count + titles.Aggregate((sentance, next) => sentance + ", " + next );
```

Comment:
This has changed the formatting as there is no longer a space after the total count

Reply:
Also, (though a micro optimisation) using aggregate here will create lots of strings in memory and then return the final string.
String.join does not do this, so can this side of the new if statement remain unchanged